File: typescript/models/Booking.ts
class Booking {
  private _id: number
  private _roomId: number
  private _checkIn: number
  private _checkOut: number

  constructor(id: number, roomId: number, checkIn: number, checkOut: number) {
    this._id = id
    this._roomId = roomId
    this._checkIn = checkIn
    this._checkOut = checkOut
  }

  get id() { return this._id }
  get roomId() { return this._roomId }
  get checkIn() { return this._checkIn }
  get checkOut() { return this._checkOut }
}

export default Booking

File: net7/Untitled/Untitled.Core/Models/Booking.cs
namespace Untitled.Core.Models;

public class Booking
{
    public int Id { get; init; }
    public int RoomId { get; init; }
    public DateTime CheckIn { get; init; }
    public DateTime CheckOut { get; init; }
}

public class UnprocessedBooking
{
    public int RoomId { get; init; }
    public DateTime CheckIn { get; init; }
    public DateTime CheckOut { get; init; }
}